### PR TITLE
<CopyClipboard/> - add new component

### PR DESCRIPTION
**Pull request by daugerdas**
### 🔦 Summary

Add a new wrapper component `<CopyClipboard/>` which uses [`useCopyClipboard`](https://github.com/wix/wix-style-react/tree/master/src/providers/useCopyClipboard) hook to copy strings to clipboard 

> CopyClipboard is a function mechanism that wraps other components to provide copy/paste functionality. Use it to copy urls, code snippets or similar. 

Description: [DSM-769](https://jira.wixpress.com/browse/DSM-769)

### CODE REVIEW

**Comment by WixBuildServer** [comment]
### 🤖 Hi, I'm #Janet!

  A deploy preview of your PR was successfully released 🖼.

  You can find it under one of the following URLs:
  * https://preview.wix-style-react.com/pr_6367

File: src/CopyClipboard/CopyClipboard.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    value,
+    onCopy,
+    resetTimeout,
+  });
+
+  return (
+    <div className={className} data-hook={dataHook}>
+      {children({
+        copyToClipboard,
+        isCopied,
```
Comment (mykas) [inline src/CopyClipboard/CopyClipboard.js:18]: I believe we should expose reset function here too

File: src/CopyClipboard/CopyClipboard.js [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  );
+};
+
+CopyClipboard.displayName = 'CopyClipboard';
+
+CopyClipboard.propTypes = {
+  /** Applied as data-hook HTML attribute that can be used in the tests */
+  dataHook: PropTypes.string,
+
+  /** Children render prop */
```
Comment (mykas) [inline src/CopyClipboard/CopyClipboard.js:30]: Please add explanation of render function and available render props here

File: src/CopyClipboard/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+
+type CopyClipboardChildrenProps = {
+  copyToClipboard: Function;
+  isCopied?: boolean;
+};
+
+export interface CopyClipboardProps {
+  value: string;
+  children(params: CopyClipboardChildrenProps): React.ClassicElement<any>;
```
Comment (mykas) [inline src/CopyClipboard/index.d.ts:10]:
```suggestion
  children(params: CopyClipboardChildrenProps): React.Element;
```

File: src/CopyClipboard/CopyClipboard.js (hunk 1 above)
Comment (daugerdas) [reply src/CopyClipboard/CopyClipboard.js:17]: yes, definitely, thank you! It is added now in dfcfd56

File: src/CopyClipboard/index.d.ts [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import * as React from 'react';
+
+type CopyClipboardChildrenProps = {
+  copyToClipboard: Function;
+  isCopied?: boolean;
```
[the remark this answers is not in the document]
Comment (daugerdas) [reply src/CopyClipboard/index.d.ts:5]: added it now in dfcfd56

File: src/CopyClipboard/CopyClipboard.js (hunk 2 above)
Comment (daugerdas) [reply src/CopyClipboard/CopyClipboard.js]: updated in be32212

**Review by mykas — APPROVED** [review]

**Pull request merged**